lint-rules.md

```diff
@@ -0,0 +1,1280 @@
+        File in_bam
+        File input_gtf
+    }
+```
+
+### `disallowed_output_name`
+
+Any output name matching these regular expressions will be flagged: `/^[oO]ut[A-Z_]/`, `/^output/i` or `/^..?$/`. It is redundant and needlessly verbose to use an output's name to specify that it is an output. Output names should be short yet descriptive. Prefixing a name with `out` or `output` adds length to the name without adding clarity or context. Additionally, names with only 2 characters can lead to confusion and obfuscates the content of an output. Output names should be at least 3 characters long.
+
+**Group**: `naming`
+
+#### Example
+
+All the below output names would be flagged:
+
+```wdl
+    input {
+        File outBam  # This name will be flagged twice, once for the 'out' prefix, and once for being camalCase when it should be snake_case
+        File out_bam
+        File output_gtf
+        String af
+    }
+```
+
+### `no_curly_commands`
+
+`command` blocks should be wrapped with arrows (`<<<` `>>>`) instead of curly brackets (`{` `}`). Certain Bash constructions cause problems with parsing the bracket notation. There are no such problems with the arrow notation.
+
+#### Example
+
+Good:
+
+```wdl
+    command <<<
+        echo "Hello, World!"
+    >>>
+```
+
+Bad:
+
+```wdl
+    command {
+        echo "Hello, World!"
+    }
+```
+
+### `mutable_container`  && `immutable_container_not_tagged`
+
+All tasks should run in an immutable container. This ensures reproducibility across time and environments. `wdl-grammar` and `wdl-ast` will look for a `:SHASUM` tag in your `container` declarations and warn if one is missing. A `sha` digest will always point to the exact same image, whereas tags are mutable. This mutability makes even versioned tags problematic when we want to ensure reproducibility.
+
+While the confidence in persistence gained by using `sha` digests for pulling container images is valuable, it comes at a cost: lack of human readability. So we enforce following `container` strings with a comment which gives a human readable name to the image being pulled. It is imperative that the `sha` digest and the human readable tag are kept in sync. It's extra overhead compared to just using a tag directly, but we consider it well worth it.
+
+Note that `container` lines are permitted to exceed the 90 character width limit.
+
+**Group**: `container`
+
+#### Example
+
+Bad:
+
+```wdl
+    runtime {
+        memory: "4 GB"
+        disks: "10 GB"
+        container: "docker://quay.io/biocontainers/htseq:2.0.4--py310h5aa3a86_0"
+        maxRetries: 1
+    }
+```
+
+Good:
+
+```wdl
+    runtime {
+        memory: "4 GB"
+        disks: "10 GB"
+        container: "docker://quay.io/biocontainers/htseq@sha256:04309f74909f7e48bc797ee5faa4e4388d7f581890c092a455d15bbcf5f6c537"  # tag: 2.0.4
+        maxRetries: 1
+    }
+```
+
+### `file_coercion`
```

Comment:
- I'm not sure whether I'd just shut this lint off or if it should be modified, but its current form disallows e.g. setting defaults that point to public cloud locations, like a reference file input, etc. which is a common use case for the coercion

Reply:
This is another great point! Thank you! When I wrote this, I really only meant to exclude **local** file paths. But you are correct, as written, I've also excluded files which should be available to any environment with an internet connection. I'll have to workshop the language to be more clear on this issue.

Reply:
That makes sense, though might be tricky to distinguish "local" paths from "global" paths when verifying. If there's a good way to do that, I think it could make a good lint as `disallow_local_file_paths` or something.